recipes/obake/meta.yaml

```
@@ -0,0 +1,90 @@
+{% set version = "0.1" %}
+
+package:
+  name: obake
+  version: {{ version }}
+
+source:
+  url: https://github.com/bluescarni/obake/archive/v{{ version }}.tar.gz
+  sha256: 946de817940421d8af8c614c289b74dc6e221ea803fc2b3dacc30e63d7dc3246
+
+build:
+  number: 0
+  skip: true   # [win and vc<14]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - cmake
+    - clang      # [win]
+    - ninja      # [win]
+
+outputs:
+  - name: obake
+    requirements:
+      host:
+        - abseil-cpp
+        - tbb-devel
+        - tbb
+        - mppp
+        - boost-cpp
+        - backtrace    # [linux]
+      run:
+        - {{ pin_compatible("mppp") }}
+        - backtrace    # [linux]
+    files:
+      - lib/libobake*{{ SHLIB_EXT }}   # [not win]
+      - lib/libobake.so.*              # [linux]
+      - lib/libobake.*.dylib           # [osx]
+      - Library/bin/obake.dll          # [win]
+    test:
+      commands:
+        - test -f $PREFIX/lib/libobake.so                                     # [linux]
+        - test -f $PREFIX/lib/libobake.dylib                                  # [osx]
+        - if not exist %PREFIX%\\Library\\bin\\obake.dll exit 1               # [win]
+
+  - name: obake-devel
+    build:
+      run_exports:
+        # SO name changes at minor rev bumps.
+        - {{ pin_subpackage('obake', max_pin='x.x') }}
+        - boost-cpp
```

```suggestion
        - {{ pin_compatible('boost-cpp', max_pin='x.x') }}
```
It's annoying that we have to use "x.x" here, but there's no way in conda-build to say, use the `run_exports` from the package.

---

Are the pins on ``mppp`` corrects in the recipe? I really don't understand how this works :(

---

Are `mp++` headers used in `obake` public headers?

---

Yes. I also did the ``run_exports`` thing in the mppp recipe, but not sure if I did it correctly:

https://github.com/conda-forge/mppp-feedstock/blob/master/recipe/meta.yaml#L15

What I wanted to convey is that libmppp.so will change its ABI (and its SO version number) at every minor x.x release.